src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java

```diff
@@ -347,5 +350,32 @@ public void onDisconnectPlayer(IChatComponent reason, CallbackInfo ci) {
         Sponge.getGame().getEventManager().post(event);
         event.getSink().sendMessage(event.getNewMessage());
     }
+    
+    @Inject(method = "processUseEntity", at = @At(value = "INVOKE", target =
+             "net/minecraft/network/play/client/C02PacketUseEntity.getAction()Lnet/minecraft/network/play/client/C02PacketUseEntity$Action;",
+             ordinal = 0), cancellable = true, locals = LocalCapture.CAPTURE_FAILSOFT)
+    public void processUseEntity(C02PacketUseEntity packetIn, CallbackInfo info, WorldServer worldserver, Entity entity, boolean flag, double d0) {
+        EntityInteractionType type = null;
+        Vector3d location = null;
+        switch (packetIn.getAction()) {
+            case ATTACK:
+                type = EntityInteractionTypes.ATTACK;
+                break;
+            case INTERACT_AT:
+                location = new Vector3d(packetIn.getHitVec().xCoord, packetIn.getHitVec().yCoord, packetIn.getHitVec().zCoord);
```

Comment:
Also, you're not setting the `type` variable in this block, should be `EntityInteractionTypes.USE`